File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def insertIntoBST(self, root, val):
        def helper(root,val):  
            if val < root.val:
                if root.left:
                    helper(root.left, val)
                else:
                    root.left = TreeNode(val)
            else:
                if root.right:
                    helper(root.right, val)               
                else:
                    root.right = TreeNode(val)
        
        if not root:
            return TreeNode(val)
        else:
            helper(root,val)
            return root

File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.c

struct TreeNode* insertIntoBST(struct TreeNode* root, int val) {
    struct TreeNode* cur = root;
    struct TreeNode* toInsert = (struct TreeNode*)malloc(sizeof(struct TreeNode));
    toInsert->val = val;
    
    if(!root){
        toInsert->left = toInsert->right = NULL;
        return toInsert;
    }
    
    while (cur) {
        if (val < cur->val) {
            if (cur->left) cur = cur->left;
            else {
                cur->left = toInsert;
                toInsert->left = toInsert->right = NULL;
                return root;
            }
        }
        else {
            if (cur->right) cur = cur->right;
            else {
                cur->right = toInsert;
                toInsert->right = toInsert->left = NULL;
                return root;
            }
        }
    }
    
    return root;


}
